src/sa.py

```diff
@@ -1,15 +1,98 @@
 import argparse
+import fasta, fastq
+import re
 
+def getSuffixes(x: memoryview):
+    """
+    Gets all suffixes from a string
+    """
+    suffixes = [x[i:] for i in range(0, len(x))] 
+    return suffixes
 
+def getTrailingNumber(s):
+    m = re.search(r'\d+$', s)
+    return (s[:m.start()], int(s[m.start():]))
+
+def lower(a: str, x: str, sa: memoryview, lo: int, hi: int, offset: int) -> int:
+    """Finds the lower bound of `a` at `offset` in the block defined by `lo:hi`."""
+    while lo < hi:  # Search in sa[lo:hi]
+        m = (lo + hi) // 2
+        if x[sa[m] + offset % len(x)] < a: # compare at column offset in sa
```

Comment:
Shouldn't this be `(sa[m] + offset) % len(x)` instead? Taking the remainder of the offset is fine and all, but that doesn't wrap around `x`. Let's say `x` is 10 long. If `sa[m]` points to the last index of `x`, 9, and I want to look at offset 3, I would get `x[9 + 3 % 10] == x[9+3] == x[12]` which is out of bounds. If, on the other hand, I have the parentheses, I would get `x[(9+3) % 10] == x[2]` which is fine. I've wrapped around `x`, but I am still in the right range, and the sentinel will handle the order.